benchmark-analyze-results.py

```
@@ -1,36 +1,48 @@
         new_data = json.load(f)
+    return old_data, new_data
+
 
-    # Извлекаем данные для графиков
+def extract_metrics(old_data, new_data):
     performance_data = []
     memory_data = []
     gc_data = []
+    allocation_data = []
 
     for old_bench, new_bench in zip(old_data, new_data):
```

Ruff (0.15.18)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)